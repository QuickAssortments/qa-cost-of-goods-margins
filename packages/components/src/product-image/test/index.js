/** @format */
/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import ProductImage from '../';

describe( 'ProductImage', () => {
	test( 'should render the passed alt prop', () => {
		const image = shallow( <ProductImage alt="testing" /> );
		expect( image ).toMatchSnapshot();
	} );

	test( 'should fallback to product alt text', () => {
		const product = {
			name: 'Test Product',
			images: [
				{
					src: 'https://i.cloudup.com/pt4DjwRB84-3000x3000.png',
					alt: 'hello world',
				},
			],
		};
		const image = shallow( <ProductImage product={ product } /> );
		expect( image ).toMatchSnapshot();
	} );

	test( 'should fallback to empty alt attribute if not passed via prop or product object', () => {
		const product = {
			name: 'Test Product',
			images: [
				{
					src: 'https://i.cloudup.com/pt4DjwRB84-3000x3000.png',
				},
			],
		};
		const image = shallow( <ProductImage product={ product } /> );
		expect( image ).toMatchSnapshot();
	} );

	test( 'should have the correct width and height', () => {
		const image = <ProductImage width={ 30 } height={ 30 } />;
		expect( image.props.width ).toBe( 30 );
		expect( image.props.height ).toBe( 30 );
	} );

	test( 'should render a product image', () => {
		const product = {
			name: 'Test Product',
			images: [
				{
					src: 'https://i.cloudup.com/pt4DjwRB84-3000x3000.png',
				},
			],
		};
		const image = shallow( <ProductImage product={ product } /> );
		expect( image ).toMatchSnapshot();
	} );

	test( 'should render a variations image', () => {
		const variation = {
			name: 'Test Variation',
			image: {
				src: 'https://i.cloudup.com/pt4DjwRB84-3000x3000.png',
			},
		};
		const image = shallow( <ProductImage product={ variation } /> );
		expect( image ).toMatchSnapshot();
	} );

	test( 'should render a placeholder image if no product images are found', () => {
		global.wcSettings.wcAssetUrl = 'https://woocommerce.com/wp-content/plugins/woocommerce/assets/';
		const product = {
			name: 'Test Product',
		};
		const image = shallow( <ProductImage product={ product } /> );
		expect( image ).toMatchSnapshot();
	} );
} );
