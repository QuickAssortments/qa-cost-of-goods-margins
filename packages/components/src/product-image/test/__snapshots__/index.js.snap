// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should fallback to empty alt attribute if not passed via prop or product object 1`] = `
<img
  alt=""
  className="woocommerce-product-image"
  height={60}
  src="https://i.cloudup.com/pt4DjwRB84-3000x3000.png"
  width={60}
/>
`;

exports[`ProductImage should fallback to product alt text 1`] = `
<img
  alt="hello world"
  className="woocommerce-product-image"
  height={60}
  src="https://i.cloudup.com/pt4DjwRB84-3000x3000.png"
  width={60}
/>
`;

exports[`ProductImage should render a placeholder image if no product images are found 1`] = `
<img
  alt=""
  className="woocommerce-product-image is-placeholder"
  height={60}
  src="https://woocommerce.com/wp-content/plugins/woocommerce/assets/images/placeholder.png"
  width={60}
/>
`;

exports[`ProductImage should render a product image 1`] = `
<img
  alt=""
  className="woocommerce-product-image"
  height={60}
  src="https://i.cloudup.com/pt4DjwRB84-3000x3000.png"
  width={60}
/>
`;

exports[`ProductImage should render a variations image 1`] = `
<img
  alt=""
  className="woocommerce-product-image"
  height={60}
  src="https://i.cloudup.com/pt4DjwRB84-3000x3000.png"
  width={60}
/>
`;

exports[`ProductImage should render the passed alt prop 1`] = `
<img
  alt="testing"
  className="woocommerce-product-image is-placeholder"
  height={60}
  src="undefinedimages/placeholder.png"
  width={60}
/>
`;
